packager/live_packager_test.cc

```diff
@@ -649,6 +666,63 @@ TEST_F(LivePackagerBaseTest, VerifyAes128WithDecryption) {
   }
 }
 
+// Test demonstrates decrypting fmp4 source with one set of keys and
+// re-encrypting using a difference encryption schema (AES-128) and different
+// set of keys. As validation the re-encryption segments are decrypted and
+// compared against decrypted segments.
+TEST_F(LivePackagerBaseTest, VerifyPrdDecryptReEncrypt) {
+  std::vector<uint8_t> init_segment_buffer =
+      ReadTestDataFile("encrypted/prd_data/init.mp4");
+  ASSERT_FALSE(init_segment_buffer.empty());
+
+  media::AesCbcDecryptor decryptor(media::kPkcs5Padding,
+                                   media::AesCryptor::kUseConstantIv);
+  ASSERT_TRUE(decryptor.InitializeWithIv(key_, iv_));
+
+  for (unsigned int i = 1; i <= 7; i++) {
+    std::string segment_num = absl::StrFormat("encrypted/prd_data/%05d.m4s", i);
+    std::vector<uint8_t> segment_buffer = ReadTestDataFile(segment_num);
+    ASSERT_FALSE(segment_buffer.empty());
+
+    SegmentData init_seg(init_segment_buffer.data(),
+                         init_segment_buffer.size());
+    SegmentData media_seg(segment_buffer.data(), segment_buffer.size());
+
+    LiveConfig live_config;
+    live_config.format = LiveConfig::OutputFormat::TS;
```

Comment:
Could we add another test for fmp4 output?  

Also, it looks like the source init segment contains `pssh` boxes, we'll need a small update to the interface to support it.  @Vadym-Kupriyanchuk could you please confirm if this is a requirement?

Reply:
yes, we will need update this pssh in case we are doing encryption, also we need to remove it if its only decryption operation.